step-templates/windows-service-stop-or-kill.json

```diff
@@ -3,15 +3,17 @@
+    "Octopus.Action.Script.ScriptBody": "$svcName = $OctopusParameters['ServiceName']\n$svcTimeout = $OctopusParameters['ServiceStopTimeout']\n\nfunction Stop-ServiceWithTimeout ([string] $name, [int] $timeoutSeconds) {\n    $timespan = New-Object -TypeName System.Timespan -ArgumentList 0,0,$timeoutSeconds\n\n    If (Get-Service $svcName -ErrorAction SilentlyContinue) {\n        if ($svc -eq $null) { return $true }\n        if ($svc.Status -eq [ServiceProcess.ServiceControllerStatus]::Stopped) { return $true }\n        $svc.Stop()\n        try {\n            Write-Host \"Stopping Service\" $svcTimeout \"Timeout\"\n            $svc.WaitForStatus([ServiceProcess.ServiceControllerStatus]::Stopped, $timespan)\n        }\n        catch [ServiceProcess.TimeoutException] {\n            Write-Host \"Timeout stopping service $($svc.Name)\"\n            return $false\n        }\n        Write-Host \"Service Sucessfully stopped\"\n        return $true\n\n    } Else {\n        Write-Host \"Service does not exist, this is acceptable. Probably the first time deploying to this target\"\n        Exit\n    }\n}\n\nWrite-Host \"Checking service\"\n\n$svcpid = (get-wmiobject Win32_Service | where{$_.Name -eq $svcName}).ProcessId\nWrite-Host \"Found PID \" + $svcpid \n\nStop-ServiceWithTimeout -name $svcName -timeoutSeconds $svcTimeout\n\nWrite-Host \"Rechecking service\"\n$svcpid = (get-wmiobject Win32_Service | where{$_.Name -eq $svcName}).ProcessId\nWrite-Host \"Found PID \" + $svcpid \n\n$service = Get-Service -name $svcName | Select -Property Status\nif($service.Status -ne \"Stopped\"){\tStart-Sleep -seconds 5 }\n\n#Check-Service process \nif($svcpid){\n    #still exists?\n    $p = get-process -id $svcpid -ErrorAction SilentlyContinue\n    if($p){\n        Write-Host \"Killing Service\"\n        Stop-Process $p.Id -force\n    }\n}",
```

Comment:
```
If (Get-Service $svcName -ErrorAction SilentlyContinue) {
    if ($svc -eq $null) { return $true } <-- This will always we true as $svc is never assigned
    ...
}
```
The first `if` condition above should be `$svc = Get-Service $svcName -ErrorAction SilentlyContinue` instead.

Reply:
you're absolutely right.